hummingbot/connector/derivative/bybit_perpetual/bybit_perpetual_derivative.py

```diff
@@ -186,6 +179,46 @@ def tracking_states(self) -> Dict[str, Any]:
     def budget_checker(self) -> PerpetualBudgetChecker:
         return self._budget_checker
 
+    async def _set_position_mode(self, position_mode: PositionMode):
+        """
+        An overwriten method from the base class that sends a request to change the position mode first
+        :param position_mode: ONEWAY or HEDGE position mode
+        """
+        mode = None
+        if position_mode == PositionMode.ONEWAY:
+            mode = CONSTANTS.POSITION_MODE_API_ONEWAY
+        elif position_mode == PositionMode.HEDGE:
+            mode = CONSTANTS.POSITION_MODE_API_HEDGE
+
+        if self._trading_pairs is not None:
+            for trading_pair in self._trading_pairs:
+                symbol = await self._trading_pair_symbol(trading_pair)
+                body_params = {"symbol": symbol, "mode": mode}
+
+                response = await self._api_request(
+                    method="POST",
+                    endpoint=CONSTANTS.SET_POSITION_MODE_URL,
+                    body=body_params,
+                    is_auth_required=True,
+                )
+
+                response_code = response["ret_code"]
+                # 0     - mode changed
+                # 30083 - mode not modified
+                if response_code not in [0, 30083]:
+                    raise IOError(f"Bybit Perpetual encountered a problem switching position mode to "
+                                  f"{position_mode} for {trading_pair}"
+                                  f" ({response['ret_code']} - {response['ret_msg']})")
+
+        super().set_position_mode(position_mode)
+
+    def set_position_mode(self, position_mode: PositionMode):
+        """
+        An overwriten method from the base class that sends a request to change the position mode first
+        :param position_mode: ONEWAY or HEDGE position mode
+        """
+        safe_ensure_future(self._set_position_mode(position_mode))
```

Comment:
Is this functionality supported by Bybit for non linear markets (inverse perpetual)?

Reply:
It should. But I have not found anything in the documentation about setting position mode for non-linear markets. The documentation doesn't correspond to the real state of their API. Maybe such URL exists but I have no way of knowing.

Reply:
I check in their web interface with my personal account and there's no option to change position mode in case of inverted perpetuals therefore I've added a check and an exception.